packages/aws-cdk-lib/aws-dynamodb/lib/table-v2.ts

```diff
@@ -674,6 +693,15 @@ export class TableV2 extends TableBaseV2 {
       ? (props.region === this.region ? this.tableOptions.resourcePolicy : props.resourcePolicy) || undefined
       : props.resourcePolicy ?? this.tableOptions.resourcePolicy;
 
+    const propTags: Record<string, string> = (props.tags ?? []).reduce((p, item) =>
+      ({ ...p, [item.key]: item.value }), {},
+    );
+
+    const tags: CfnTag[] = Object.entries({
+      ...this.tags.tagValues(),
+      ...propTags,
+    }).map(([k, v]) => ({ key: k, value: v }));
+
```

Comment:
can you use TagManager.renderTags function instead of handling it manually. You can check [this example](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts#L174) for how to use it